doc/source/whatsnew/v1.3.0.rst

```diff
@@ -312,6 +312,8 @@ Numeric
 
 Conversion
 ^^^^^^^^^^
+- Bug in :meth:`DataFrame.to_dict` with ``orient='records'`` now returns python native datetime objects for datetimelike columns (:issue:`21256`)
```

Comment:
this is no longer true rigth (e.g. this is not fixed in this PR)

Reply:
yeah i'll get rid of this